File: backend/src/main/kotlin/com/example/backend/models/Pharmacy.kt
package com.example.backend.models

import jakarta.persistence.*

@Entity
@Table(name = "pharmacy")
class Pharmacy {
  @Id
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "nom", length = 64)
  var nom: String? = null

  @Column(name = "telephone", nullable = false, length = 128)
  var telephone: String? = null

  @Column(name = "adresse", nullable = false, length = 128)
  var adresse: String? = null

  @Column(name = "logo", nullable = false, length = 64)
  var logo: String? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "CodePostal_id")
  var codePostal: CodePostal? = null

  @Column(name = "slogan", nullable = false, length = 128)
  var slogan: String? = null

  @Column(name = "docteur", nullable = false, length = 128)
  var docteur: String? = null

  @Column(name = "contribuable", nullable = false, length = 128)
  var contribuable: String? = null
}

File: backend_py/pharmaPython/app/models/pharmacy.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from app.api.v1.db import Base

class Pharmacy(Base):
  __tablename__ = "pharmacy"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True)
  nom = Column(String(64))
  telephone = Column(String(128), nullable=False)
  adresse = Column(String(128), nullable=False)
  logo = Column(String(64), nullable=False)
  code_postal_id = Column(Integer, ForeignKey("code_postal.id"))
  code_postal = relationship("CodePostal", lazy="select")
  slogan = Column(String(128), nullable=False)
  docteur = Column(String(128), nullable=False)
  contribuable = Column(String(128), nullable=False)
